core/include/detray/tools/navigator.hpp

```diff
@@ -216,280 +288,309 @@ class navigator {
-        inspector_type _inspector = {};
+        /// The inspector type of this navigation engine
+        inspector_type _inspector;
 
-        /** Index of a object (surface/portal) if is reached, otherwise invalid
-         */
+        /// Index of a object (surface/portal) if is reached, otherwise invalid
         dindex _object_index = dindex_invalid;
 
-        /**  The navigation status */
-        navigation_status _status = e_unknown;
+        /// The navigation status
+        navigation::status _status = navigation::e_unknown;
 
-        /** The on object tolerance - permille */
+        /// The on object tolerance - permille
         scalar _on_object_tolerance = 1e-3;
 
-        /** The navigation trust level */
-        navigation_trust_level _trust_level = e_no_trust;
+        /// The navigation trust level
+        navigation::trust_level _trust_level = navigation::e_no_trust;
 
-        /** Volume we are currently navigating in */
-        dindex _volume_index = dindex_invalid;
+        /// Volume we are currently navigating in
+        dindex _volume_index = 0;
     };
 
     DETRAY_HOST_DEVICE
     navigator(const detector_t &d) : _detector(&d) {}
 
-    template <typename track_t>
-    DETRAY_HOST_DEVICE inline intersection update_next_candidate(
-        state &navigation, const track_t &track) const {
-
-        dindex obj_idx = navigation.next()->index;
-        auto sfi =
-            intersect(track, _detector->surfaces()[obj_idx],
-                      _detector->transform_store(), _detector->mask_store());
-        sfi.index = obj_idx;
-        (*navigation.next()) = sfi;
-
-        return sfi;
+    /// Alias call to update for readbility
+    template <typename stepper_state_t>
+    DETRAY_HOST_DEVICE bool init(state &navigation,
+                                 stepper_state_t &stepping) const {
+        // Establish navigation status before the first step
+        return update(navigation, stepping);
     }
 
     /** Navigation status() call which established the current navigation
      *  information.
      *
-     * @tparam stepper_state_t type of stepper state
+     * @tparam track_t type of the track (including context)
      *
      * @param navigation [in, out] is the navigation state object
-     * @param stepping the stepper state
+     * @param stepping stepper state that contains the track
      *
      * @return a heartbeat to indicate if the navigation is still alive
      **/
     template <typename stepper_state_t>
-    DETRAY_HOST_DEVICE inline bool status(state &navigation,
+    DETRAY_HOST_DEVICE inline bool update(state &navigation,
                                           stepper_state_t &stepping) const {
         bool heartbeat = true;
 
-        const auto &track = stepping();
-
-        if (navigation.trust_level() >= e_high_trust) {
-            while (not navigation.is_exhausted()) {
+        if (navigation.candidates().capacity() != 0)
```

Comment:
Is it correct? candidates buffer in CUDA is always non-zero

Reply:
Hm, Ok, I didn't know that. But I am also not sure why that is still there, so I will remove it